src/integration-test/java/com/commercetools/sync/integration/sdk2/commons/utils/CartDiscountITUtils.java

```diff
@@ -0,0 +1,285 @@
+    QueryUtils.queryAll(
+            ctpClient.cartDiscounts().get(),
+            cartDiscounts -> {
+              CompletableFuture.allOf(
+                      cartDiscounts.stream()
+                          .map(cartDiscount -> deleteCartDiscount(ctpClient, cartDiscount))
+                          .map(CompletionStage::toCompletableFuture)
+                          .toArray(CompletableFuture[]::new))
+                  .join();
+            })
+        .toCompletableFuture()
+        .join();
+  }
+
+  private static CompletionStage<CartDiscount> deleteCartDiscount(
+      final ProjectApiRoot ctpClient, final CartDiscount cartDiscount) {
+    return ctpClient
+        .cartDiscounts()
+        .delete(cartDiscount)
+        .execute()
+        .thenApply(ApiHttpResponse::getBody);
+  }
+
+  /**
+   * Deletes all cart discounts from CTP projects defined by {@code CTP_SOURCE_CLIENT} and {@code
+   * CTP_TARGET_CLIENT}.
+   */
+  public static void deleteCartDiscountsFromTargetAndSource() {
+    deleteCartDiscounts(CTP_SOURCE_CLIENT);
+    deleteCartDiscounts(CTP_TARGET_CLIENT);
+  }
+
+  public static void populateSourceProject() {
+    ensureCartDiscountCustomType(
+        OLD_CART_DISCOUNT_TYPE_KEY, Locale.ENGLISH, OLD_CART_DISCOUNT_TYPE_NAME, CTP_SOURCE_CLIENT);
+
+    final CartDiscountDraft draft1 =
+        CartDiscountDraftBuilder.of(CART_DISCOUNT_DRAFT_1)
+            .custom(
+                CustomFieldsDraftBuilder.of()
+                    .type(
+                        typeResourceIdentifierBuilder ->
+                            typeResourceIdentifierBuilder.key(OLD_CART_DISCOUNT_TYPE_KEY))
+                    .fields(createCustomFieldsJsonMap())
+                    .build())
+            .build();
+    final CartDiscountDraft draft2 =
+        CartDiscountDraftBuilder.of(CART_DISCOUNT_DRAFT_2)
+            .custom(
+                CustomFieldsDraftBuilder.of()
+                    .type(
+                        typeResourceIdentifierBuilder ->
+                            typeResourceIdentifierBuilder.key(OLD_CART_DISCOUNT_TYPE_KEY))
+                    .fields(createCustomFieldsJsonMap())
+                    .build())
+            .build();
+
+    CompletableFuture.allOf(
+            CTP_SOURCE_CLIENT.cartDiscounts().create(draft1).execute(),
+            CTP_SOURCE_CLIENT.cartDiscounts().create(draft2).execute())
+        .join();
+  }
+
+  public static Type ensureCartDiscountCustomType(
+      @Nonnull final String typeKey,
+      @Nonnull final Locale locale,
+      @Nonnull final String name,
+      @Nonnull final ProjectApiRoot ctpClient) {
+
+    return createTypeIfNotAlreadyExisting(
+        typeKey, locale, name, List.of(ResourceTypeId.CART_DISCOUNT), ctpClient);
+  }
+
+  public static CartDiscount populateTargetProject() {
+    ensureCartDiscountCustomType(
+        OLD_CART_DISCOUNT_TYPE_KEY, Locale.ENGLISH, OLD_CART_DISCOUNT_TYPE_NAME, CTP_TARGET_CLIENT);
+
+    final CartDiscountDraft draft1 =
```

Comment:
Looks like this is same draft as used in `populateSourceProject()`, maybe create a constant field?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1025/commits/4bdf37e01c099c494a151eb7ed73f264b3e6f4c1